osu.Game/Screens/Play/BreakPeriodsTrackOverlay.cs

```diff
@@ -0,0 +1,248 @@
+            {
+                arrows.PlayWarning();
+                startArrowsHasBeenShown = true;
+            }
+
+            var currentBreak = breaks[currentBreakIndex];
+
+            if (!isBreak)
+            {
+                if (currentTime > currentBreak.StartTime)
+                {
+                    isBreak = true;
+
+                    if (currentBreak.HasEffect)
+                    {
+                        iconHasBeenShown = false;
+                        arrowsHasBeenShown = false;
+                        iconAppearTime = currentTime + (currentBreak.EndTime - currentBreak.StartTime) / 2;
+                        BreakIn?.Invoke();
+                    }
+                }
+            }
+            else
+            {
+                // Show icon depends on HP
+                if (currentTime > iconAppearTime && !iconHasBeenShown && currentBreak.HasPeriodResult)
+                {
+                    if (health < 0.3)
+                    {
+                        resultIcon.Icon = FontAwesome.fa_close;
+                        sampleFail.Play();
+                    }
+                    else
+                    {
+                        resultIcon.Icon = FontAwesome.fa_check;
+                        samplePass.Play();
+                    }
+
+                    resultIcon.FadeTo(1);
+                    Delay(100);
+                    Schedule(() => resultIcon.FadeTo(0));
+                    Delay(100);
+                    Schedule(() => resultIcon.FadeTo(1));
+                    Delay(1000);
+                    Schedule(() => resultIcon.FadeTo(0, 200));
+
+                    iconHasBeenShown = true;
+                }
+
+                // Show warning arrows
+                if (currentBreak.EndTime - currentTime < arrows_appear_offset && !arrowsHasBeenShown)
+                {
+                    if (currentBreak.HasEffect)
+                        arrows.PlayWarning();
+
+                    arrowsHasBeenShown = true;
+                }
+
+                // Exit from break
+                if (currentBreak.EndTime - currentTime < fade_duration)
+                {
+                    if (currentBreak.HasEffect)
+                        BreakOut?.Invoke();
+
+                    currentBreakIndex++;
+                    isBreak = false;
+                }
+            }
+        }
+
+        private class ArrowsOverlay : Container
+        {
+            private const int appear_duration = 140;
+            private const int margin_vertical = 120;
+            private const int margin_horizontal = 90;
+
+            private readonly Container content;
+
+            public ArrowsOverlay()
+            {
+                RelativeSizeAxes = Axes.Both;
+                Add(content = new Container
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Alpha = 0,
+                    Children = new Drawable[]

```

Comment:
We are doing the same thing in HudOverlay. I guess, we don't wanna hide the container itself, only the content inside this container.